### PR TITLE
Refactory to support Waggle Dance changes

### PR DESCRIPTION
**Pull request by AnanaMJ**
* added localhost to SshSettings
* moved `checkNotNull` and `checkIsTrue` from waggle-dance to this project's `Preconditions` class
* related to https://github.com/HotelsDotCom/waggle-dance/issues/103

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,14 @@
 ## TBD
+### Added
+* SshSettings has a new parameter can be assigned: localhost.
```
Comment (abhimanyugupta07) [inline CHANGELOG.md]: has a new parameter can be assigned --> has a new parameter 'that' can be assigned

Comment (massdosage) [inline CHANGELOG.md]: can be -> that can be

**Review by massdosage — APPROVED** [review]
Minor comment on changelog, the rest looks good.

File: CHANGELOG.md (hunk 1 above)
Comment (AnanaMJ) [reply CHANGELOG.md]: done :)

**Review by abhimanyugupta07 — APPROVED** [review]

**Pull request merged**